python_modules/libraries/dagster-k8s/dagster_k8s/executor.py

```diff
@@ -237,9 +237,10 @@ def _get_container_context(
         user_defined_k8s_config = get_user_defined_k8s_config(
             step_handler_context.step_tags[step_key]
         )
-
+        step_context = step_handler_context.get_step_context(step_key)
+        op_name = step_context.step.op_name
         per_op_override = UserDefinedDagsterK8sConfig.from_dict(
-            self._per_step_k8s_config.get(step_key, {})
+            self._per_step_k8s_config.get(op_name, {})
```

Comment:
@gibsondan - could this break in some other context?

Reply:
I don't think these will ever differ in a case where you would not prefer the op name, so i think this change is correct